### PR TITLE
Test archive

### LINKED ISSUE #79: Test Archiving Data
### Task

- [x] Test archive.py
- [x] Test clean_national_rail.py
- [x] Test clean_real_time_trains.py
- [x] Test db_connection.py

### User Story
As a developer, I want to test that the functionality of Archive works as expected and reduces the chances of bugs to occur in the future.

### PR DESCRIPTION
**Pull request by FarihaChoudhury**
Tests for archiving

closes #79

### CODE REVIEW

File: archive/test_archive.py [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+    @patch("db_connection.get_connection")
+    def test_get_cursor(self, mock_get_connection: connection):
+        """ Tests get_cursor returns cursor. """
+
+        mock_connection = MagicMock()
+        mock_get_connection.return_value = mock_connection
+        db_cursor = get_cursor(mock_connection)
+
+        mock_connection.cursor.assert_called_once_with(
+            cursor_factory=RealDictCursor)
```
Comment (Zhi-704) [inline archive/test_archive.py:57]: Why are you still using realdictcursor here?

Comment (FarihaChoudhury) [reply archive/test_archive.py:57]: I DONT KNOW OKAY

File: archive/test_archive.py [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+        mock_cursor.fetchall.assert_called_once()
+        self.assertEqual(result, None)
+
+    @patch("db_connection.get_cursor")
+    def test_delete_waypoint(self, mock_get_cursor):
+        """ Test delete waypoint successful """
+
+        mock_cursor = MagicMock()
+        mock_get_cursor.return_value.__enter__.return_value = mock_cursor
+        mock_conn = MagicMock()
```
Comment (Zhi-704) [inline archive/test_archive.py:213]: Good call

File: archive/test_archive.py (hunk 1 above)
Comment (FarihaChoudhury) [reply archive/test_archive.py:57]: so I can fetch data as a dict instead of a tuple

Comment (Zhi-704) [reply archive/test_archive.py:57]: Fine by me

**Review by Zhi-704 — APPROVED** [review]
From the goodness of my heart

**Pull request merged**